platform/darwin/src/MGLStyle.mm

```diff
@@ -534,6 +540,25 @@ - (void)insertLayer:(MGLStyleLayer *)layer aboveLayer:(MGLStyleLayer *)sibling {
     [self didChangeValueForKey:@"layers"];
 }
 
+#pragma mark - Layer retain/release management
+
+- (void)addToManagedLayers:(MGLStyleLayer*)layer {
```

Comment:
This needs to be invoked from `[MGLStyle layerFromMBGLLayer]` as well, to capture internally created style layers.